src/shady-css/common.js

```diff
@@ -0,0 +1,38 @@
+let matcher = {
+  whitespace: /\s/,
+  whitespaceGreedy: /(\s+)/g,
+  commentOpen: /^\/\*/,
+  commentClose: /\*\/$/,
+  boundary: /[\(\)\{\}\[\]'"@;:\s\.#+>~\^]|\/\*/,
+  // TODO: "Block boundary" is a bad name for this character set:
+  blockBoundary: /[;\{\}]/,
+  combinatorBoundary: /[\{]/,
+  propertyBoundary: /^[;:]/,
+  // TODO: This also delimits atRules
+  propertyDelimiter: /^[;\}]/,
+  propertyNameTrail: /:$/,
+  stringBoundary: /['"]/
+};
+
+let tokenType = {
```

Comment:
these could move into lexer.js. the parser depends on the lexer and will pick these up there.